dexter/articles.py

```diff
@@ -22,13 +25,44 @@
 
 @app.route('/articles/<id>')
 @login_required
-@roles_accepted('monitor')
+@roles_accepted('monitor', 'fdi')
 def show_article(id):
     document = Document.query.get_or_404(id)
-
     if request.args.get('format') == 'places-json':
         return jsonify(DocumentPlace.summary_for_docs([document]))
 
+    if current_user.has_role('fdi'):
+
+        exists = Investment.query.filter_by(doc_id=id).first() is not None
+        if not exists:
```

Comment:
I suggest only creating the Investment object when the FDI analysis is first submitted. Otherwise, if a user that has the FDI role (perhaps William or you) just looks at an article, it'll create an entry in the database. That'll then be included in the database.